ledger/statecommit/committer.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package statecommit
+
+import (
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/merklearray"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// stateUpdate represents a single insert/update or delete operation
+type stateUpdate struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Key     []byte `codec:"k"`
+	Value   []byte `codec:"v"`
+	Deleted bool   `codec:"d"`
```

Comment:
You don't want to use `nil` Value as delete? I think we do that quite a bit elsewhere.